embassy-stm32/build.rs

```diff
@@ -103,6 +103,70 @@ fn main() {
         }
     });
 
+    // ========
+    // Generate FLASH regions
+    let mut flash_regions = TokenStream::new();
+    let flash_memory_regions = METADATA
+        .memory
+        .iter()
+        .filter(|x| x.kind == MemoryRegionKind::Flash && x.settings.is_some());
+    for region in flash_memory_regions.clone() {
+        let region_name = format_ident!("{}", get_flash_region_name(region.name));
+        let base = region.address as usize;
+        let size = region.size as usize;
+        let settings = region.settings.as_ref().unwrap();
+        let erase_size = settings.erase_size as usize;
+        let write_size = settings.write_size as usize;
+        let erase_value = settings.erase_value;
+
+        flash_regions.extend(quote! {
+            #[allow(non_camel_case_types)]
+            pub struct #region_name(());
+        });
+
+        flash_regions.extend(quote! {
+            impl crate::flash::FlashRegion for #region_name {
+                const BASE: usize = #base;
+                const SIZE: usize = #size;
+                const ERASE_SIZE: usize = #erase_size;
+                const WRITE_SIZE: usize = #write_size;
+                const ERASE_VALUE: u8 = #erase_value;
+            }
+        });
+    }
+
+    let (fields, inits): (Vec<TokenStream>, Vec<TokenStream>) = flash_memory_regions
+        .map(|f| {
+            let region_name = get_flash_region_name(f.name);
+            let field_name = format_ident!("{}", region_name.to_lowercase());
+            let field_type = format_ident!("{}", region_name);
+            let field = quote! {
+                pub #field_name: #field_type
+            };
+            let init = quote! {
+                #field_name: #field_type(())
+            };
+
+            (field, init)
+        })
+        .unzip();
+
+    flash_regions.extend(quote! {
+        pub struct FlashRegions {
+            #(#fields),*
+        }
+
+        impl FlashRegions {
+            pub(crate) const fn take() -> Self {
```

Comment:
`take` sounds like it's removing the original (e.g. from an Option). I'd name this `new`.

Reply:
Renamed to new...